DataPlotly/core/plot_factory.py

```diff
@@ -110,26 +111,39 @@ def add_source_field_or_expression(field_or_expression):
             return None, False, {field_or_expression}
 
         x_expression, x_needs_geom, x_attrs = add_source_field_or_expression(self.settings.properties['x_name']) if \
-        self.settings.properties[
+            self.settings.properties[
             'x_name'] else (None, False, set())
         y_expression, y_needs_geom, y_attrs = add_source_field_or_expression(self.settings.properties['y_name']) if \
-        self.settings.properties[
+            self.settings.properties[
             'y_name'] else (None, False, set())
         z_expression, z_needs_geom, z_attrs = add_source_field_or_expression(self.settings.properties['z_name']) if \
-        self.settings.properties[
+            self.settings.properties[
             'z_name'] else (None, False, set())
         additional_info_expression, additional_needs_geom, additional_attrs = add_source_field_or_expression(
             self.settings.layout['additional_info_expression']) if self.settings.layout[
             'additional_info_expression'] else (None, False, set())
 
         attrs = set().union(x_attrs, y_attrs, z_attrs, additional_attrs)
 
-        request = QgsFeatureRequest().setSubsetOfAttributes(attrs, self.source_layer.fields())
+        request = QgsFeatureRequest()
+
+        if self.settings.properties['feature_subset_query']['active']:
+            if 'expression' in self.settings.properties['feature_subset_query']:
```

Comment:
Would be better to re-use add_source_field_or_expression here, otherwise you're basically just rewriting this logic (and missing the check if the geometry is required)